libs/sdk-ui-kit/src/@ui/UiFocusTrap/UiFocusTrap.tsx

```diff
@@ -0,0 +1,125 @@
+// (C) 2025 GoodData Corporation
+
+import React, { useEffect, useRef } from "react";
+
+/**
+ * @internal
+ */
+export interface UiFocusTrapProps {
+    children: React.ReactNode;
+    onDeactivate?: () => void;
+    returnFocusTo?: React.RefObject<HTMLElement>;
+}
+const focusableElementsSelector = [
+    // Interactive form elements
+    'button:not(:disabled):not([aria-disabled="true"])',
+    'input:not(:disabled):not([aria-disabled="true"])',
+    'select:not(:disabled):not([aria-disabled="true"])',
+    'textarea:not(:disabled):not([aria-disabled="true"])',
+
+    // Links and areas
+    "a[href]",
+    "area[href]",
+
+    // Custom elements with tabindex
+    '[tabindex]:not([tabindex="-1"]):not(:disabled):not([aria-disabled="true"])',
+
+    // Media with controls
+    "audio[controls]",
+    "video[controls]",
+
+    // Editable content
+    '[contenteditable]:not([contenteditable="false"])',
+].join(",");
+
+const getFocusableElements = (element?: HTMLElement) => {
+    const focusableElements = element?.querySelectorAll<HTMLElement>(focusableElementsSelector);
+    const firstElement = focusableElements?.[0];
+    const lastElement = focusableElements?.[focusableElements.length - 1];
+    return { focusableElements, firstElement, lastElement };
+};
+
+/**
+ * @internal
+ */
+export const UiFocusTrap: React.FC<UiFocusTrapProps> = ({ children, onDeactivate, returnFocusTo }) => {
+    const trapRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        const getNextElement = (focusableElements, firstElement, lastElement, shiftKey: boolean) => {
+            const elements = Array.from(focusableElements);
+            const currentIndex = elements.indexOf(document.activeElement as HTMLElement);
+
+            let nextElement: HTMLElement | undefined;
+            if (shiftKey) {
+                // Shift + Tab - moving backwards
+                nextElement = currentIndex <= 0 ? lastElement : elements[currentIndex - 1];
```

Comment:
Minor: You can inline these by just do `return currentIndex ... ;` for both branches, no need for local variable.